noxfile.py

```diff
@@ -87,6 +87,8 @@ def env_vars_for_test(
     if toplevel_lang == "vhdl" and gpi_interface is not None:
         e["VHDL_GPI_INTERFACE"] = gpi_interface
 
+    # Fail on warnings in cocotb and test modules
+    e["PYTHONWARNINGS"] = "error:::cocotb,error:::test_.*" 
```

Comment:
This doesn't appear to be working, we should be seeing failures for throwing `FutureWarning` in `cocotb.scheduler`.

From [the docs](https://docs.python.org/3/library/warnings.html#the-warnings-filter), it says...
>module is a string containing a regular expression that the start of the fully qualified module name must match, case-sensitively.

So `cocotb` should match `cocotb.scheduler`... needs more testing.